libraries/apollo-normalized-cache-sqlite-incubating/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/sql/SqlNormalizedCache.kt

```diff
@@ -17,66 +18,101 @@ class SqlNormalizedCache internal constructor(
     private val recordDatabase: RecordDatabase,
 ) : NormalizedCache() {
 
-  override fun loadRecord(key: String, cacheHeaders: CacheHeaders): Record? {
-    val record = try {
-      recordDatabase.select(key)
-    } catch (e: Exception) {
-      // Unable to read the record from the database, it is possibly corrupted - treat this as a cache miss
-      apolloExceptionHandler(Exception("Unable to read a record from the database", e))
-      null
-    }
-    if (record != null) {
-      if (cacheHeaders.hasHeader(EVICT_AFTER_READ)) {
-        recordDatabase.delete(key)
+  // A lock is only needed if there is a nextCache
```

Comment:
Are we 100% certain of that? SQLDelight queries being thread safe no matter what underlying driver is used?

Reply:
Not 100% certain alas. At least on native (MacOS) I made a small test that starts 2 threads, 1st starts a transaction, 2nd one also, I could see that the 2nd one was blocked until the 1st one finished. I am not sure if this is due to a lock somewhere in the driver, or if SQLite itself handles this. What I think I understand from [here](https://sqlite.org/isolation.html) is that SQLite handles this itself, but that's not 100% clear to me tbh.